_pytest/debugging.py

```diff
@@ -97,6 +97,11 @@ def _enter_pdb(node, excinfo, rep):
         tw.sep(">", "captured stderr")
         tw.line(captured_stderr)
 
+    captured_logs = rep.caplog
+    if len(captured_logs) > 0:
+        tw.sep(">", "captured logs")
```

Comment:
Please check the option inside the `if` block:

```python
if len(captured_logs) > 0:
    if not config.getoption('no_print_logs'):
        tw.sep(">", "captured logs")
        tw.line(captured_logs)
```

Doing the alternative:

```python
if len(captured_logs) > 0 and not config.getoption('no_print_logs'):
```

Is incorrect because it will raise if the logging plugin is disabled with `-p no:logging`.

Reply:
Which reminds me, we don't have a test to ensure pytest doesn't blow up if `logging` is disabled, could you add one as well? Make sure to emit logging messages inside the sub-test, such a test would caught the error I mention above.

Reply:
Yesterday I discovered that a `--show-capture` option has been added recently (see #3176). IMO this option should supersede the `--no-print-logs` option (see #3233 and #3234, where I removed the `--no-print-logs` option in favor of `--show-capture`). The PR #3234 is based on this PR. Please have a look at this PR and tell me what you think.

Reply:
Makes sense. I'll wait until you merge #3234, and then I'll update this PR accordingly?

Reply:
As I said in https://github.com/pytest-dev/pytest/pull/3234#issuecomment-366559644, we should not really remove the `--no-print-logs` option, we should deprecate it first.

Reply:
But I guess that we don't need to use `print_logs` in debugging.py. We can use `show_capture` instead.

Reply:
@brianmaissy I guess that we can close this PR and merge #3234 instead, which contains your commit https://github.com/pytest-dev/pytest/pull/3234/commits/069f32a8c452e7dbd4d10d4da2c142dd24bb5953.

Reply:
sounds good. do you want to take the test which checks with `-p no:logging` or a variation of it?

Reply:
Yes, we should add it there or in a new PR.